libs/coin-modules/coin-evm/src/preload.ts

```diff
@@ -1,83 +1,101 @@
-import axios from "axios";
+import { AxiosError } from "axios";
 import { log } from "@ledgerhq/logs";
-import {
-  addTokens,
-  convertERC20,
-  convertBEP20,
-  listTokensForCryptoCurrency,
-} from "@ledgerhq/cryptoassets/tokens";
-import { getEnv } from "@ledgerhq/live-env";
-import { CryptoCurrency } from "@ledgerhq/types-cryptoassets";
-import { BEP20Token, ERC20Token } from "@ledgerhq/cryptoassets/types";
+import { addTokens, convertERC20 } from "@ledgerhq/cryptoassets/tokens";
 import {
   tokens as tokensByChainId,
   hashes as embeddedHashesByChainId,
 } from "@ledgerhq/cryptoassets/data/evm/index";
+import { ERC20Token } from "@ledgerhq/cryptoassets/types";
+import { CryptoCurrency } from "@ledgerhq/types-cryptoassets";
+import { fetchTokensFromCALService } from "@ledgerhq/cryptoassets/crypto-assets-importer/fetch/index";
 import { getCALHash, setCALHash } from "./logic";
 
 export const fetchERC20Tokens: (
   currency: CryptoCurrency,
-) => Promise<ERC20Token[] | BEP20Token[] | null> = async currency => {
+) => Promise<ERC20Token[] | null> = async currency => {
   const { ethereumLikeInfo } = currency;
+  const { chainId } = ethereumLikeInfo || {};
 
-  const latestCALHash = getCALHash(currency);
-  const embeddedHash =
-    embeddedHashesByChainId[ethereumLikeInfo?.chainId as keyof typeof embeddedHashesByChainId];
-  const embeddedTokens = tokensByChainId[
-    ethereumLikeInfo?.chainId as keyof typeof tokensByChainId
-  ] as ERC20Token[];
+  if (!chainId) return null;
 
-  const url = `${getEnv("DYNAMIC_CAL_BASE_URL")}/evm/${ethereumLikeInfo?.chainId || 0}/erc20.json`;
-  const tokens = await axios
-    .get<ERC20Token[]>(url, {
-      headers: {
-        "If-None-Match": latestCALHash,
+  const embeddedHash = embeddedHashesByChainId[chainId as keyof typeof embeddedHashesByChainId];
+  const latestCALHash = getCALHash(currency);
+  try {
+    const { tokens, hash } = await fetchTokensFromCALService(
+      {
+        chainId: chainId,
+        standard: chainId === 56 ? "bep20" : "erc20",
```

Comment:
is the standard necessary here?
I try 
https://crypto-assets-service.api.ledger.com/v1/tokens?output=id,exchange_app_config_serialized,exchange_app_signature&chain_id=56&standard=bep20

and

https://crypto-assets-service.api.ledger.com/v1/tokens?output=id,exchange_app_config_serialized,exchange_app_signature&chain_id=56

Both of them work.

Reply:
Yes unfortunately, the standard is mandatory to limit the "chainIds" to only EVMs. 
If you're trying the same thing with chainId 1, you'll see we're getting a lot of other currencies:
https://crypto-assets-service.api.ledger.com/v1/tokens?output=id,exchange_app_config_serialized,exchange_app_signature&chain_id=1